docs/performance/remote-filesystem/amazon-s3.md

```diff
@@ -0,0 +1,396 @@
+                'credentials' => $credentials,
+            ];
+
+            return $app['aws']->createS3($options);
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function boot(Application $app)
+    {
+    }
+}
+```
+
+Optionally, you may with to construct your own `\Aws\Credentials\Credentials`
+object, and pass that to `CredentialProvider::fromCredentials()`, e.g.
+
+```php
+        $app['aws.s3'] = $app->share(function ($app) {
+            $accessKey = 'your access key from Amazon';
+            $secretKey = 'your secret key from Amazon';
+            $credentials = CredentialProvider::fromCredentials(new Credentials($accessKey, $secretKey));
+            $options = [
+                'version' => '2006-03-01',
+                'credentials' => $credentials,
+            ];
+
+            return $app['aws']->createS3($options);
+        });
+```
+
+<p class="note"><strong>Note:</strong> This particular exacmple should be
+considered conceptual, to simply illustrate the basic requirements.
+<br>
+Saving of thngs like secrect keys to code that gets committed to version
+control is bad security practice.
+</p>
+
+
+### Assets
+
+In order for functionality, like Twig's `{{ asset() }}` function to know about
+the variation in location of assets that is return to the requesting client,
+you need to next override the desired package in the package group with a
+`Symfony\Component\Asset\UrlPackage` object that defines your base Amazon S3
+bucket URL.
+
+This is achived by extending `$app['asset.packages']`, and adding your specific
+`UrlPackage` object to he desired package pool name.
+
+
+#### Example
+
+This is an example of the `src/Provider/AssetServiceProvider.php` file.
+
+```php
+<?php
+
+namespace MySiteApp\Provider;
+
+use Silex\Application;
+use Silex\ServiceProviderInterface;
+use Symfony\Component\Asset\UrlPackage;
+
+class AssetServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function register(Application $app)
+    {
+        $app['asset.packages'] = $app->share(
+            $app->extend(
+                'asset.packages',
+                function ($packages, $app) {
+                    $package = new UrlPackage(
+                        'http://example.s3-website.eu-central-1.amazonaws.com',
+                        $app['asset.version_strategy']('files'),
+                        $app['asset.context']
+                    );
+                    $packages->addPackage('files', $package);
+
+                    return $packages;
+                }
+            )
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function boot(Application $app)
+    {
+    }
+}
+```
+
+
+The first parameter for `UrlPackage` is the base URL or your Amazon S3 bucket.
+
+The second parameter is the version strategy applicable to the package, the
+value in the final parentheses **must** be the name of the package you're
+replacing.
```

Comment:
It must be the name of the filesystem you are replacing. But it should match the package name as well, or the fs asset plugin will break.